spec/services/rate_creator_spec.rb

```diff
@@ -1,13 +1,27 @@
 require 'rails_helper'
 
 RSpec.describe RateCreator do
-  let!(:submission) { FactoryGirl.create(:submission) }
-  let!(:user) { FactoryGirl.create(:user) }
-  subject { described_class.build(1, submission.id, user.id) }
-
-  it 'creates a new rate' do
-    result = subject.call
-    expect(Rate.where(value: 1, submission_id: submission.id, user_id: user.id).first).not_to be_nil
-    expect(result.success).to be true
+  let(:submission) { FactoryGirl.create(:submission) }
+  let(:user) { FactoryGirl.create(:user) }
+  let(:value) { 1 }
+  subject { described_class.build(value, submission.id, user.id) }
+
+  context 'when no rates' do
```

Comment:
I'd write something more explicit here like "when the user hasn't rated the submission yet", similarly in the second context